tests/test_main.py

```diff
@@ -37,7 +38,28 @@ def test_version(arg, capsys):
 
 
 def test_dummy_run(capsys):
-    run_main()
+    run_main(code=0)
     stdout, stderr = capsys.readouterr()
     assert stdout == ""
     assert stderr == ""
+
+
+@patch("retasc.__main__.validate_rule")
+def test_validate_rule(mock_validate_rule):
+    run_main("validate-rule", "any_valid_rule.yaml", code=0)
+    mock_validate_rule.assert_called_once_with("any_valid_rule.yaml")
+
+
+@patch("retasc.__main__.validate_rule")
+def test_validate_invalid_rule(mock_validate_rule):
```

Comment:
Please test also if the `main()` prints expected messages to the user. Not printing anything could be confusing.

Reply:
This still does not check the expected error message, just that the exit code is 1 which can mean other things.